### PR TITLE
Integrated facebook messenger

### LINKED ISSUE #7672: [Integration] Integrate Facebook Messenger with MindsDB
The goal of this issue is to build a new handler that will utilize the Facebook Messenger API to connect MindsDB to Facebook Messenger.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building Application Handler docs](https://docs.mindsdb.com/contribute/app-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices.

* :book:  [Facebook Messenger docs](https://developers.facebook.com/docs/messenger-platform/overview)

* :point_right: [Twitter Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/twitter_handler): This existing handler connects MindsDB with the Twitter API, allowing users to automate tasks related to posting and analyzing tweets. You can use the Twitter Handler as a reference for building the Slack API integration.

*  :point_right:  [Binance API Handler](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/binance_handler): Another example of a successful MindsDB integration to the Binance API to make trade data available to use for model training and predictions.

## Next Steps :male_detective: :female_detective: 
 
-   You can familiarize yourself with the Facebook Messenger API.
-   Create a new handler that connects MindsDB to Facebook Messenger.
-   Implement the core features mentioned above.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the new Facebook Messenger Integration handler.

## :loudspeaker: Additional rewards :trophy:

Each App integration brings :five: :zero:  : points for SWAG and entry into the draw for a  :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by Vishalk91-4**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #7672

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by Vishalk91-4** [comment]
@ZoranPandovski, could you please review this PR for Hacktoberfest

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thanks. Please make sure you have all of the informations added to the README file and you check all of the steps in the PR checklist

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by Vishalk91-4** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Vishalk91-4** [comment]
@ZoranPandovski, made the changes

File: mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+        return result
+
+    def get_columns(self):
+        return [
+            'id', 'created_time', 'message', 'from_id', 'from_name', 'to_id', 'to_name',
+        ]
+
+
+def insert(self, query):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py:121]: Why this function is outside of the class?

File: mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+class MessagesTable(APITable):
+    def select(self, query: ast.Select) -> Response:
+        conditions = extract_comparison_conditions(query.where)
+        params = {}
+        filters = []
+
+        for op, arg1, arg2 in conditions:
+            if op == 'text':
+                if arg1 == 'message':
+                    client = Messager('<access_token>')
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py:36]: Do we need to provide the correct value here?

File: mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+                    client.send_text(arg2, 'Recipient ID')
+            elif op == 'image':
+                if arg1 == 'message':
+                    send_api = SendApi('<page_access_token>')
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py:40]: Do we need to provide the correct value here?

File: mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+                    send_api.send_local_video(arg2, 'Recipient ID')
+            elif op == 'file':
+                if arg1 == 'message':
+                    send_api = SendApi('<page_access_token>')
+                    send_api.send_local_file(arg2, 'Recipient ID')
+            elif op == 'quick_reply':
+                if arg1 == 'message':
+                    client = Messager('<access_token>')
+                    client.send_quick_replies('Recipient ID', arg2, [
+                        QuickReply("Option 1", "PAYLOAD_1"),
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py:58]: Isn't this a hardcoded payload?

File: mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+            raise Exception('Failed to update persistent menu: ' + response.text)
+
+
+class FacebookMessengerHandler(APIHandler):
+    def __init__(self, name=None, **kwargs):
+        super().__init__(name)
+        args = kwargs.get('connection_data', {})
+        self.connection_args = {}
+        handler_config = Config().get('facebookmessenger_handler', {})
+        for k in ['access_token']:
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/facebook_messenger_handler/messenger_handler.py:174]: Shouldn't this be a page_access_token ?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

**Comment by ZoranPandovski** [comment]
@Vishalk91-4 This doesn't work, please check the comments above and make sure you provide screenshots or video of working integrations and include tests

**Comment by chandrevdw31** [comment]
Hey @Vishalk91-4 we are coming to a closing with the Hacktoberfest and would like to have this PR merged so that we can gift swag credits to participants. Kindly make above changes and DM me on Slack

**Pull request closed without merging**